crates/solver/src/solver/optimizing_solver.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{
+    settlement::Settlement,
+    settlement_post_processing::PostProcessing,
+    solver::{Auction, Solver},
+};
+use anyhow::Result;
+use ethcontract::Account;
+use gas_estimation::GasPrice1559;
+use model::auction::AuctionId;
+use shared::http_solver::model::AuctionResult;
+use std::sync::Arc;
+
+/// A wrapper for solvers that applies a set of optimizations to all the generated settlements.
+pub struct OptimizingSolver {
+    pub inner: Arc<dyn Solver>,
+    pub post_processing_pipeline: Arc<dyn PostProcessing>,
+}
+
+#[async_trait::async_trait]
+impl Solver for OptimizingSolver {
+    async fn solve(&self, auction: Auction) -> Result<Vec<Settlement>> {
+        let gas_price = GasPrice1559 {
+            base_fee_per_gas: auction.gas_price,
+            max_fee_per_gas: auction.gas_price,
+            max_priority_fee_per_gas: 0.,
```

Comment:
This might be the indicator that we need to change the gas price in solver `Auction` from `f64` to `GasPrice1559` in some future PR.